internals/reminders.py

```diff
@@ -89,18 +117,30 @@ def get_template_data(self, **kwargs):
         current_milestone_info)
     email_tasks = build_email_tasks(
         features_to_notify, self.SUBJECT_FORMAT, self.EMAIL_TEMPLATE_PATH,
-        current_milestone_info)
+        current_milestone_info, self.ALLOW_ESCALATION, self.ESCALATION_CHECK)
     notifier.send_emails(email_tasks)
 
     message =  f'{len(email_tasks)} email(s) sent or logged.'
     logging.info(message)
+
+    if hasattr(self, 'cleanup'):
```

Comment:
Rather than checking if there is a cleanup method defined, why not define one here that simply has a `pass` statement, and then override that method in subclasses when needed?

Also, the name `cleanup` is not ideal because it is not cleaning up resources that are left in some messy state.  Maybe try something like `post_process` or let it be more specific to the one way that it is actually used now with a name like `update_counts`.

Reply:
This is better, and I've implemented it this way. 🙂